packages/parse-mapping-lookup/src/ast.ts

```diff
@@ -0,0 +1,111 @@
+import debugModule from "debug";
+const debug = debugModule("parse-mapping-lookup:ast");
+
+/*
+ * Expression
+ */
+
+export interface Expression {
+  kind: "expression";
+  root: Identifier;
+  pointer: Pointer;
+}
+
+export const expression = (options: {
+  root: Identifier;
+  pointer: Pointer;
+}): Expression => {
+  const { root, pointer } = options;
+  return {
+    kind: "expression",
+    root,
+    pointer
+  };
+};
+
+/*
+ * Identifier
+ */
+
+export interface Identifier {
+  kind: "identifier";
+  name: string;
+}
+
+export const identifier = (options: { name: string }): Identifier => {
+  const { name } = options;
+  return {
+    kind: "identifier",
+    name
+  };
+};
+
+/*
+ * Pointer
+ */
+
+export interface Pointer {
+  kind: "pointer";
+  path: Step[];
+}
+
+export type Step = Access | Lookup;
+
+export const pointer = (options: { path: Step[] }): Pointer => {
+  const { path } = options;
+  return {
+    kind: "pointer",
+    path
+  };
+};
+
+/*
+ * Lookup
+ */
+
+export interface Lookup {
+  kind: "lookup";
+  property: Identifier;
+}
+
+export const lookup = (options: { property: Identifier }): Lookup => {
+  const { property } = options;
+  return {
+    kind: "lookup",
+    property
+  };
+};
+
+/*
+ * Access
+ */
+
+export interface Access {
+  kind: "access";
+  index: Literal;
```

Comment:
So here's one more thing -- what about enums?  Those aren't *literals* per se.  One possibility would be to allow not just literals here, but I think the more appropriate route (although admittedly hackier) would be to allow "enum literals" as a type of literal as well (in addition to the other types I suggested above).

Reply:
What forms do enum literals take? Is it always `<identifier>.<identifier>`? Or you can do `<identifier>.<identifier>.<identifier>` also, right?

Reply:
Yup, you can do both!

Reply:
(Ended up implementing this one tonight also.)